extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -0,0 +1,121 @@
+package io.quarkus.kubernetes.deployment;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import io.ap4k.Session;
+import io.ap4k.SessionWriter;
+import io.ap4k.docker.annotation.EnableDockerBuild;
+import io.ap4k.docker.generator.DefaultEnableDockerBuildGenerator;
+import io.ap4k.kubernetes.annotation.KubernetesApplication;
+import io.ap4k.kubernetes.generator.DefaultKubernetesApplicationGenerator;
+import io.ap4k.processor.SimpleFileWriter;
+import io.ap4k.project.Project;
+import io.quarkus.deployment.BuildInfo;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.BuildInfoBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.kubernetes.spi.KubernetesPortBuildItem;
+
+class KubernetesProcessor {
+
+    @Inject
+    BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer;
+
+    @Inject
+    BuildProducer<FeatureBuildItem> featureProducer;
+
+    @BuildStep
+    public void build(BuildInfoBuildItem buildInfoBuildItem,
+            KubernetesConfig kubernetesConfig,
+            KubernetesPortBuildItem kubernetesPortBuildItem) {
+
+        // The resources that ap4k's execution will result in, will later-on be written
+        // by quarkus in the 'wiring-classes' directory
+        // The location is needed in order to properly support s2i build triggering
+        final Path rootPath = Paths.get(buildInfoBuildItem.getBuildInfo().getWiringClassesDir());
+        if (rootPath.toString().isEmpty()) { //this is done to prevent ap4k from failing when the necessary input has not been provided (during tests for example)
+            return;
+        }
+        final Path ap4kRoot = rootPath.resolve("META-INF/ap4k");
```

Comment:
Since `ap4k` is working behind the scenes, maybe this needs to be `META-INF/kubernetes` or something.

Reply:
I am not sure this will work with what ap4k already does internally, I'll have to check

Reply:
For vanilla Kubernetes it works just fine, so I'll recheck when I implement openshift + s2i